### ISSUE TITLE
ER_TOO_LONG_IDENT on LinkToAnotherRecord on MySQL

### ISSUE THREAD

**Issue by DrMint** [nocodb/nocodb#913]
**Please enter the following details**

Node: **v12.22.6**
Arch: **x64**
Platform: **linux**
Docker: **true**
Database: **mysql2**
ProjectOnRootDB: **true**
RootDB: **mysql2**
PackageVersion: **0.84.1**

**Steps To Reproduce**
1. Create a table "MyFirstTable"
2. Create another table "MySecondTableWithLongerName"
3. Create a ManyToMany LinkToAnotherRecord column from MyFirstTable to MySecondTableWithLongerName

From there, either it shows a success message, or there are both a success and error message. In either cases,  no new column was created.

Here the logs

```
db-nocodb-1   | Error: alter table `nc_vqxf__my_first_table` add constraint `nc_vqxf__my_first_table_nc_vqxf__my_second_table_with_longer_name_id_foreign` foreign key (`nc_vqxf__my_second_table_with_longer_name_id`) references `nc_vqxf__my_second_table_with_longer_name` (`id`) on update NO ACTION on delete NO ACTION - Identifier name 'nc_vqxf__my_first_table_nc_vqxf__my_second_table_with_longer_name_id_foreign' is too long
db-nocodb-1   |     at Packet.asError (/usr/src/app/node_modules/mysql2/lib/packets/packet.js:722:17)
db-nocodb-1   |     at Query.execute (/usr/src/app/node_modules/mysql2/lib/commands/command.js:28:26)
db-nocodb-1   |     at Connection.handlePacket (/usr/src/app/node_modules/mysql2/lib/connection.js:456:32)
db-nocodb-1   |     at PacketParser.onPacket (/usr/src/app/node_modules/mysql2/lib/connection.js:85:12)
db-nocodb-1   |     at PacketParser.executeStart (/usr/src/app/node_modules/mysql2/lib/packet_parser.js:75:16)
db-nocodb-1   |     at Socket.<anonymous> (/usr/src/app/node_modules/mysql2/lib/connection.js:92:25)
db-nocodb-1   |     at Socket.emit (events.js:314:20)
db-nocodb-1   |     at Socket.EventEmitter.emit (domain.js:483:12)
db-nocodb-1   |     at addChunk (_stream_readable.js:297:12)
db-nocodb-1   |     at readableAddChunk (_stream_readable.js:272:9) {
db-nocodb-1   |   code: 'ER_TOO_LONG_IDENT',
db-nocodb-1   |   errno: 1059,
db-nocodb-1   |   sqlState: '42000',
db-nocodb-1   |   sqlMessage: "Identifier name 'nc_vqxf__my_first_table_nc_vqxf__my_second_table_with_longer_name_id_foreign' is too long",
db-nocodb-1   |   sql: 'alter table `nc_vqxf__my_first_table` add constraint `nc_vqxf__my_first_table_nc_vqxf__my_second_table_with_longer_name_id_foreign` foreign key (`nc_vqxf__my_second_table_with_longer_name_id`) references `nc_vqxf__my_second_table_with_longer_name` (`id`) on update NO ACTION on delete NO ACTION'
db-nocodb-1   | }
```

**Expected behavior**
Creating LinkToAnotherRecord columns on two tables with slightly long names should be possible on MySQL. I don't know if that's a hard limitation on the technology, in which case it should probably be explained somewhere in the doc. In any case, I think those names are not particularly long, I often use NameOfTable_Translations for translated content.

**Comment by o1lab** [comment]
We've to think of an uuid suffixed foreign key index creation

**Comment by pranavxc** [comment]
Fixed in `0.84.7`

**Comment by AddoSolutions** [comment]
I am getting this in version `0.204.4`

```
ALTER TABLE `nc_fa0x___eCommerce Item` 
    ADD
    COLUMN `nc_fa0x___eCommerce Transaction_id4` int(11) UNSIGNED NULL;
╔══════════════════════════════════════════════════════════════════════════════╗║                                                                              ║║   EXCEPTION OCCURED!! in KnexClient @ indexCreate                            ║║   - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -    ║║   - - - - - - - - - - - - - - - - - - - - - -                                ║║   MESSAGE:                                                                   ║║   alter table `nc_fa0x___eCommerce Item` add index `nc_fa0x___ecommerce      ║║   item_nc_fa0x___ecommerce transaction_id4_index`(`nc_fa0x___eCommerce       ║║   Transaction_id4`) - Identifier name 'nc_fa0x___ecommerce                   ║║   item_nc_fa0x___ecommerce transaction_id4_index' is too long                ║║                                                                              ║║   - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -    ║║   - - - - - - - - - - - - - - - - - - - - - -                                ║║   CODE:                                                                      ║║   ER_TOO_LONG_IDENT                                                          ║║                                                                              ║║   - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -    ║║   - - - - - - - - - - - - - - - - - - - - - -                                ║║   STACK:                                                                     ║║   Error: alter table `nc_fa0x___eCommerce Item` add index                    ║║   `nc_fa0x___ecommerce item_nc_fa0x___ecommerce                              ║║   transaction_id4_index`(`nc_fa0x___eCommerce Transaction_id4`) -            ║║   Identifier name 'nc_fa0x___ecommerce item_nc_fa0x___ecommerce              ║║   transaction_id4_index' is too long                                         ║║   at Packet.asError                                                          ║║   (/usr/src/app/node_modules/mysql2/lib/packets/packet.js:728:17)            ║║   at Query.execute                                                           ║║   (/usr/src/app/node_modules/mysql2/lib/commands/command.js:29:26)           ║║   at Connection.handlePacket                                                 ║║   (/usr/src/app/node_modules/mysql2/lib/connection.js:481:34)                ║║   at PacketParser.onPacket                                                   ║║   (/usr/src/app/node_modules/mysql2/lib/connection.js:97:12)                 ║║   at PacketParser.executeStart                                               ║║   (/usr/src/app/node_modules/mysql2/lib/packet_parser.js:75:16)              ║║   at Socket.<anonymous>                                                      ║║   (/usr/src/app/node_modules/mysql2/lib/connection.js:104:25)                ║║   at Socket.emit (node:events:518:28)                                        ║║   at Socket.emit (node:domain:488:12)                                        ║║   at addChunk (node:internal/streams/readable:559:12)                        ║║   at readableAddChunkPushByteMode (node:internal/streams/readable:510:3)     ║║                                                                              ║║   - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -    ║║   - - - - - - - - - - - - - - - - - - - - - -                                ║║                                                                              ║║                                                                              ║╚══════════════════════════════════════════════════════════════════════════════╝
Error: alter table `nc_fa0x___eCommerce Item` add index `nc_fa0x___ecommerce item_nc_fa0x___ecommerce transaction_id4_index`(`nc_fa0x___eCommerce Transaction_id4`) - Identifier name 'nc_fa0x___ecommerce item_nc_fa0x___ecommerce transaction_id4_index' is too long
    at Packet.asError (/usr/src/app/node_modules/mysql2/lib/packets/packet.js:728:17)
    at Query.execute (/usr/src/app/node_modules/mysql2/lib/commands/command.js:29:26)
    at Connection.handlePacket (/usr/src/app/node_modules/mysql2/lib/connection.js:481:34)
    at PacketParser.onPacket (/usr/src/app/node_modules/mysql2/lib/connection.js:97:12)
    at PacketParser.executeStart (/usr/src/app/node_modules/mysql2/lib/packet_parser.js:75:16)
    at Socket.<anonymous> (/usr/src/app/node_modules/mysql2/lib/connection.js:104:25)
    at Socket.emit (node:events:518:28)
    at Socket.emit (node:domain:488:12)
    at addChunk (node:internal/streams/readable:559:12)
    at readableAddChunkPushByteMode (node:internal/streams/readable:510:3) {
  code: 'ER_TOO_LONG_IDENT',
  errno: 1059,
  sqlState: '42000',
  sqlMessage: "Identifier name 'nc_fa0x___ecommerce item_nc_fa0x___ecommerce transaction_id4_index' is too long",
  sql: 'alter table `nc_fa0x___eCommerce Item` add index `nc_fa0x___ecommerce item_nc_fa0x___ecommerce transaction_id4_index`(`nc_fa0x___eCommerce Transaction_id4`)'
}
node:internal/process/promises:289
            triggerUncaughtException(err, true /* fromPromise */);
            ^
Error: alter table `nc_fa0x___eCommerce Item` add index `nc_fa0x___ecommerce item_nc_fa0x___ecommerce transaction_id4_index`(`nc_fa0x___eCommerce Transaction_id4`) - Identifier name 'nc_fa0x___ecommerce item_nc_fa0x___ecommerce transaction_id4_index' is too long
    at Packet.asError (/usr/src/app/node_modules/mysql2/lib/packets/packet.js:728:17)
    at Query.execute (/usr/src/app/node_modules/mysql2/lib/commands/command.js:29:26)
    at Connection.handlePacket (/usr/src/app/node_modules/mysql2/lib/connection.js:481:34)
    at PacketParser.onPacket (/usr/src/app/node_modules/mysql2/lib/connection.js:97:12)
    at PacketParser.executeStart (/usr/src/app/node_modules/mysql2/lib/packet_parser.js:75:16)
    at Socket.<anonymous> (/usr/src/app/node_modules/mysql2/lib/connection.js:104:25)
    at Socket.emit (node:events:518:28)
    at Socket.emit (node:domain:488:12)
    at addChunk (node:internal/streams/readable:559:12)
    at readableAddChunkPushByteMode (node:internal/streams/readable:510:3) {
  code: 'ER_TOO_LONG_IDENT',
  errno: 1059,
  sqlState: '42000',
  sqlMessage: "Identifier name 'nc_fa0x___ecommerce item_nc_fa0x___ecommerce transaction_id4_index' is too long",
  sql: 'alter table `nc_fa0x___eCommerce Item` add index `nc_fa0x___ecommerce item_nc_fa0x___ecommerce transaction_id4_index`(`nc_fa0x___eCommerce Transaction_id4`)'
}
```
